docs/adr/0015-macros-and-compositions.md

```
@@ -0,0 +1,429 @@
+SecRule ARGS "@detectSQLi" "id:942100,phase:2,block,..."
+```
+
+The compiled output has no notion of "macro" — only the expanded expressions.
+
+### HCL vs Custom Surface
+
+The macro concept is the same in both language bases (Phase 0 / ADR-0009), but the
+surface differs:
+
+**Custom parser:**
+```
```

markdownlint-cli2 (0.22.1)

[warning] Fenced code blocks should be surrounded by blank lines

(MD031, blanks-around-fences)